app/services/UserService.py

```diff
@@ -19,15 +25,64 @@ async def modify_user(user_id: int, model: PydanticUserModify):
 
     if model.mail and await UserInDB.all().filter(mail=model.mail).count() != 0:
         raise HTTPException(status_code=409, detail=CommonErrorMessages.MAIL_ALREADY_USED)
-    
+
     if model.login and await UserInDB.all().filter(login=model.login).count() != 0:
         raise HTTPException(status_code=409, detail=CommonErrorMessages.LOGIN_ALREADY_USED)
 
     try:
-        user_to_modify.update_from_dict(model.model_dump(exclude={"password", "password_confirm"}, exclude_none=True)) # type: ignore
+        user_to_modify.update_from_dict(
+            model.model_dump(exclude={"password", "password_confirm"}, exclude_none=True))  # type: ignore
         if model.password is not None:
             user_to_modify.hash = UserInDB.get_password_hash(model.password)
         await user_to_modify.save()
 
     except ValueError as e:
         raise HTTPException(status_code=409, detail=str(e)) from e
+
+
+async def create_user(model: PydanticUserCreate):
+    #We check if the login or mail are already used
+
+    if await UserInDB.filter(login=model.login).exists():
+        raise LoginAlreadyUsedException
+
+    if await UserInDB.get(mail=model.mail).exists():
+        raise MailAlreadyUsedException
+
+    #If no password mentionned in the body, we generate one
+    if model.password is None:
```

Comment:
If you allow the users to create their password, please add a password_confirmation field to make sure they avoid any typos.